Source/Particles/Deposition/CurrentDeposition.H

```diff
@@ -197,34 +197,49 @@ void doEsirkepovDepositionShapeN (const amrex::Real * const xp,
 
             // --- Get particle quantities
             const amrex::Real gaminv = 1.0/std::sqrt(1.0 + uxp[ip]*uxp[ip]*clightsq
-                                                     + uyp[ip]*uyp[ip]*clightsq
-                                                     + uzp[ip]*uzp[ip]*clightsq);
+                                                         + uyp[ip]*uyp[ip]*clightsq
+                                                         + uzp[ip]*uzp[ip]*clightsq);
 
             // wqx, wqy wqz are particle current in each direction
             const amrex::Real wq = q*wp[ip];
             const amrex::Real wqx = wq*invdtdx;
-#if (AMREX_SPACEDIM == 3)
+#if (defined WARPX_DIM_3D)
             const amrex::Real wqy = wq*invdtdy;
 #endif
             const amrex::Real wqz = wq*invdtdz;
 
             // computes current and old position in grid units
+#if (defined WARPX_DIM_RZ)
+            const amrex::Real r_new = std::sqrt(xp[ip]*xp[ip] + yp[ip]*yp[ip]);
+            const amrex::Real r_old = std::sqrt((xp[ip] - dt*uxp[ip]*gaminv)*(xp[ip] - dt*uxp[ip]*gaminv) +
+                                                (yp[ip] - dt*uyp[ip]*gaminv)*(yp[ip] - dt*uyp[ip]*gaminv));
+            const amrex::Real x_new = (r_new - xmin)*dxi;
+            const amrex::Real x_old = (r_old - xmin)*dxi;
+#else
             const amrex::Real x_new = (xp[ip] - xmin)*dxi;
             const amrex::Real x_old = x_new - dtsdx0*uxp[ip]*gaminv;
-#if (AMREX_SPACEDIM == 3)        
+#endif
+#if (defined WARPX_DIM_3D)        
             const amrex::Real y_new = (yp[ip] - ymin)*dyi;
             const amrex::Real y_old = y_new - dtsdy0*uyp[ip]*gaminv;
 #endif
             const amrex::Real z_new = (zp[ip] - zmin)*dzi;
             const amrex::Real z_old = z_new - dtsdz0*uzp[ip]*gaminv;
 
+#if (defined WARPX_DIM_RZ)
+            const amrex::Real theta = std::atan2(yp[ip], xp[ip]);
+            const amrex::Real vy = (-uxp[ip]*std::sin(theta) + uyp[ip]*std::cos(theta))*gaminv;
```

Comment:
Could we avoid calculating `atan2` and `sin`/`cos` by simply doing:
```const amrex::Real vy = (-uxp[ip]*yp[ip]+ uyp[ip]*xp[ip])/r_new*gaminv;```

Reply:
Ok. I was being lazy and trying to avoid the messy looking code guarding against r==0.